src/Slot.ts

```diff
@@ -18,106 +50,210 @@ export type Unsubscribe = () => void
  * The slot can also be subscribed to, by using the `on` property.
  */
 export interface Slot<RequestData=null, ResponseData=void> {
-
-    // Make the Slot callable: this is how an event is triggered
     // TODO: Find a solution to make it possible to omit the requestData as
     // optional only when explicitly typed as such by the client.
-    (requestData: RequestData): Promise<ResponseData>
-    on: (handler: Handler<RequestData, ResponseData>) => Unsubscribe
-    lazy: (connect: () => void, disconnect: () => void) => Unsubscribe
+
+    // Trigger the slot with explicit param
+    (param: string, requestData: RequestData): Promise<ResponseData>,
+
+    // Trigger the slot with default param
+    (requestData: RequestData): Promise<ResponseData>,
+
+    // Listen to events sent through the slot on explicit param
+    on(param: string, handler: Handler<RequestData, ResponseData>): Unsubscribe,
```

Comment:
I could imagine scenarios where it would make sense to imagine passing more than just a string and declaring a real interface for the `param`:
- a client wanting to listen to changes on `todo`s of a certain type and created after a certain date
- a client launching a search (with multiple filters) and listening to the far-end to see if any changes appear on that search
- ...

A generalized solution would be to add another generic parameter `SlotParameter` to type this `param`. The value of the param would need to be serializable.
This would require to _consumer_ and the _event emitter_ to share this interface, which would bring more explicitness than the `string` type than is quite enigmatic I think.

What do you think ?

I have started a branch that uses `object-hash` on the param values and serializes it using `JSON.stringify` when needed. Not ready for a PR but good starting point for a discussion if you want!

Reply:
Yes I have thought of this.
My take:
- this is something for the future and not this PR.
- if someone really want it now, they can indeed already serialize some JSON configuration themselves.

Reply:
For your third point: this is the reason why I would use `object-hash`, which computes the same hash for `{ param1: true, param2: 'test' }` and `{ param2: "test", param1: true }`